src/canbus/elm327.cpp

```diff
@@ -170,18 +211,40 @@ std::string elm327::_read()
 {
     char buf[1];
     std::string str;
+    
+    QTimer* bluetooth_watchdog = new QTimer(this);
```

Comment:
this creates a new timer everytime without destroying it?

Reply:
or wait... ICANBus is of type QObject and i think passing the parent to the timer should take care of that

Reply:
Good catch here (I need to stop coding at late hours)

I think I'll still need to clean up the timer - setting the parent (which should be done, regardless) will only cleanup the Timer when the elm327 object is destroyed (I think)

Reply:
yep as long as the elm object sticks around so will all those timers